test/e2e/storage/flexvolume.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"path"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/generated"
+)
+
+const (
+	sshPort                = "22"
+	examplesDir            = "examples/volumes/flexvolume/"
+	defaultVolumePluginDir = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec"
+	// On gce, root may be read-only and controller-manager containerized. Assume
+	// controller-manager has started with --flex-volume-plugin-dir equal to this
+	// (see cluster/gce/config-test.sh and function mount-master-pd)
+	gceMasterVolumePluginDir = "/mnt/disks/master-pd/srv/kubernetes/kubelet-plugins/volume/exec"
+)
+
+// testFlexVolume tests that a client pod using a given flexvolume driver
+// successfully mounts it and runs
+func testFlexVolume(driver string, cs clientset.Interface, config framework.VolumeTestConfig, f *framework.Framework, clean bool) {
+	tests := []framework.VolumeTest{
+		{
+			Volume: v1.VolumeSource{
+				FlexVolume: &v1.FlexVolumeSource{
+					Driver: "k8s/" + driver,
+				},
+			},
+			File: "index.html",
+			// Must match content of examples/volumes/flexvolume/dummy(-attachable) domount
+			ExpectedContent: "Hello from flexvolume!",
```

Comment:
it could be even better to make the flex driver name include something generated (it is slightly easier too). e.g. test instance abcd installs driver dummy-abcd, test qwer dummy-qwer, so test instance will be evident from that. And parallel or subsequent tests won't conflict